src/conn/conn_layered.c

```diff
@@ -1111,6 +1140,9 @@ __wti_disagg_conn_config(WT_SESSION_IMPL *session, const char **cfg, bool reconf
 
         /* Follower step-up. */
         if (reconfig && !was_leader && leader) {
+            WT_WITH_CHECKPOINT_LOCK(session, ret = __disagg_abandon_checkpoint(session));
```

Comment:
My understanding is that `__disagg_abandon_checkpoint` only has an effect in tests with PALM.

Without PALM, it's a no-op.

Does this mean there's a behavioral difference between real production work and tests?
Is this the right place to introduce test-only functionality?

Since this is for testing only, would it make sense to pull out this method to API for tests?

Reply:
It also makes me think that we want to hold the checkpoint lock here and we could waste some time waiting if the underlying function is not defined.

Reply:
+1, one extra lock might not be a big thing but in production, it's a 100% no-op under a lock.